File: TourneyTrack/interface/src/main/java/org/tourneytrack/intf/dto/data/CompetitionDto.java
package org.tourneytrack.intf.dto.data;

import lombok.Data;

import java.util.Date;
import java.util.List;

@Data
public class CompetitionDto {
    private Long id;
    private String name;
    private String description;
    private UserDto gameMaster;
    private List<UserDto> participants;
    private Date deadline;
    private boolean inProgress;
    private List<RuleSetDto> ruleSets;
}

File: TourneyTrackFrontend/tourneytrack/src/app/shared/models/competition.dto.ts
import { UserDto } from './user.dto';
import { RuleSetDto } from './rule-set.dto';

export interface CompetitionDto {
  _expanded?: string;
  id: number;
  name: string;
  description: string;
  gameMaster: UserDto;
  participants: UserDto[];
  deadline: string;       // ISO dátum string
  inProgress: boolean;
  ruleSets: RuleSetDto[];
}
